src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -45,23 +45,51 @@ auto read_pin(
  * must be absolute.
  */
 struct extent_to_write_t {
-  laddr_t addr = L_ADDR_NULL;
+  enum class type_t {
+    DATA,
+    ZERO,
+    EXISTING,
+  };
+
+  type_t type;
+  laddr_t addr;
   extent_len_t len;
   std::optional<bufferlist> to_write;
-  std::optional<paddr_t> existing_paddr = std::nullopt;
+  std::optional<paddr_t> existing_paddr;
```

Comment:
/// non-nullopt if and only if type == EXISTING

Reply:
done